src/src/com/microsoft/aad/adal/CacheKey.java

```diff
@@ -155,6 +155,14 @@ public static String createMultiResourceRefreshTokenKey(AuthenticationRequest it
         return createCacheKey(item.getAuthority(), item.getResource(), item.getClientId(), true,
                 cacheUserId);
     }
+    
+    public static String createFamilyRefreshTokenKey(final AuthenticationRequest authRequest, final String userId) {
+        if (authRequest == null) {
+            throw new IllegalArgumentException("authentication request is null");
+        }
+        
+        return createCacheKey(authRequest.getAuthority(), null, null, true, userId);
```

Comment:
I would keep it as a string. However the server decides to give it to us in the future it will come in a JSON blob and therefore have to be representable as a string. I'm using "foci-<what-comes-in-the-foci-property-from-the-server>" i.e. "foci-1" as my client ID on FRTs.